tests/components/mqtt/test_light_template.py

```diff
@@ -320,9 +326,284 @@
+    )
+    mqtt_mock.async_publish.reset_mock()
+
+    # Set half brightness
+    await common.async_turn_on(hass, "light.test", brightness=128)
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,128,,,--", 0, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+
+    # Half brightness - no scaling of RGB values sent over MQTT
+    await common.async_turn_on(
+        hass, "light.test", rgb_color=[0, 255, 128], white_value=40
+    )
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,,,40,0-255-128", 0, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+
+    # Half brightness - normalization but no scaling of RGB values sent over MQTT
+    await common.async_turn_on(
+        hass, "light.test", rgb_color=[0, 32, 16], white_value=40
+    )
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,,,40,0-255-127", 0, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+
+
+async def test_effect(hass, mqtt_mock):
+    """Test effect sent over MQTT in optimistic mode."""
+    with assert_setup_component(1, light.DOMAIN):
+        assert await async_setup_component(
+            hass,
+            light.DOMAIN,
+            {
+                light.DOMAIN: {
+                    "platform": "mqtt",
+                    "schema": "template",
+                    "effect_list": ["rainbow", "colorloop"],
+                    "name": "test",
+                    "command_topic": "test_light_rgb/set",
+                    "command_on_template": "on,{{ effect }}",
+                    "command_off_template": "off",
+                    "qos": 0,
+                }
+            },
+        )
+
+    state = hass.states.get("light.test")
+    assert state.state == STATE_OFF
+    assert state.attributes.get(ATTR_SUPPORTED_FEATURES) == 44
+
+    await common.async_turn_on(hass, "light.test")
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,", 0, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_ON
+    assert not state.attributes.get("effect")
+
+    await common.async_turn_on(hass, "light.test", effect="rainbow")
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,rainbow", 0, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_ON
+    assert state.attributes.get("effect") == "rainbow"
+
+    await common.async_turn_on(hass, "light.test", effect="colorloop")
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,colorloop", 0, False
+    )
+    state = hass.states.get("light.test")
+    assert state.state == STATE_ON
+    assert state.attributes.get("effect") == "colorloop"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5199018)